contrib/internal/httptrace/httptrace.go

```diff
@@ -108,17 +130,26 @@ func getClientIP(r *http.Request) netaddr.IP {
 		}
 		return netaddr.IP{}
 	}
+	matches := []string{}
+	var matchedIP netaddr.IP
 	for _, hdr := range ipHeaders {
 		if v := r.Header.Get(hdr); v != "" {
+			matches = append(matches, hdr)
 			if ip := check(v); ip.IsValid() {
-				return ip
+				matchedIP = ip
 			}
 		}
 	}
+	if len(matches) == 1 {
+		return matchedIP, nil
+	}
```

Comment:
If there is exactly one IP header, but it is not valid or global, this will return `netaddr.IP{}, nil`. That seems incorrect.

Reply:
The new refactored function has a validity check